### ISSUE TITLE
Idea: module config settings and documentation for date decay functions

### ISSUE THREAD

**Issue by goldenapples** [humanmade/altis-enhanced-search#128]
The default date decay function has very limited use and functionality, and overriding it requires hooking into mutiplle filters.

Basically, the built in date decay adds a number between 0 (for infinitely old posts) to 1 (for posts published in the current week) to a post's search score. (For comparison, search scores which I'm looking at can be in the range of 30-80 for exact phrase matches in title, tags, and content.) 

This "decay" is enough to make sure that posts with exactly equal weight are listed with the most recent posts first. It doesn't do more than this, though: for example if there's a post from two years ago which contains a search term in the title and the content, it'll appear above a post from the current day which contains that same search term only in the content. 
  
On a recent project, I tried hacking around on this to get results which matched my (and the client's) expectations a little better, and ended up going with a "multiply" decay function, so that score results were multiplied by the function representing the "staleness" of the content.

```php
	/*
	 * Filter post date weighting in search results to prioritize more recent
	 * results, while still returning full matches before fuzzy matches.
	 *
	 * ElasticPress defaults are to boost results by _adding_ the result
	 * of the age decay function to the relevancy score. This is fairly
	 * meaningless, because the decay function returns a value between 0 and 1,
	 * and the search score numbers are in the range of 30-80, depending on how
	 * many areas the search query is found in the resource.
	 *
	 * phpcs:disable Generic.Formatting.DisallowMultipleStatements.SameLine
	 * phpcs:disable Generic.Functions.OpeningFunctionBraceKernighanRitchie.ContentAfterBrace
	 */
	// Weigh posts from the past 30 days equally (default is '14d').
	add_filter( 'epwr_scale', function () { return '30d'; } );
	// For each std dev of age, posts lose 10% of relevance (default is .25, or losing 75%).
	add_filter( 'epwr_decay', function () { return .9; } );
	// Multiply this boost score by results rather than adding it (default is 'add').
	add_filter( 'epwr_boost_mode', function () { return 'multiply'; } );
```
It'd be nice to expose these variables - ["scale", "decay", and "boost_mode"](https://github.com/10up/ElasticPress/blob/53003720b8937f26f2157ed41329dca173d49d9a/includes/classes/Feature/Search/Search.php#L468-L501) - in the Altis module config, in the same way we allow setting the "field_boost" multiplier for each field.

**Comment by roborourke** [comment]
Sounds good, and relatively low effort. Thanks for the detailed write up Than!

**Comment by roborourke** [comment]
Fixed in #131 - thanks @goldenapples !
